TheMovieDb/HomeViewController.swift

```diff
@@ -7,27 +7,60 @@
 
 import UIKit
 
-class HomeViewController: UIViewController, UICollectionViewDelegate {
+final class HomeViewController: UICollectionViewController {
     var viewMoreButton = UIButton()
-    var collectionView: UICollectionView!
+
     let service = NetworkManager(urlSession: URLSession.shared)
+    static let categoryHeaderUpcoming = "Upcoming"
+    let categoryHeaderUpcomingId = "Upcoming"
     var movies: [Movie] = [] {
         didSet {
             collectionView.reloadData()
         }
     }
+
+    init() {
+        super.init(collectionViewLayout: HomeViewController.createLayout())
```

Comment:
call a method instead of creating static or Factory pattern can be used here, this just to avoid exposing static elements